features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/TreeViewTraversal.kt

```diff
@@ -115,6 +123,35 @@ internal class TreeViewTraversal(
     private fun isHidden(view: View): Boolean =
         view.getTag(R.id.datadog_hidden) == true
 
+    @UiThread
+    private fun updateTouchOverrideAreas(view: View) {
+        val touchPrivacy = view.getTag(R.id.datadog_touch_privacy)
+
+        if (touchPrivacy != null) {
+            val locationOnScreen = IntArray(2)
+
+            // this will always have size >= 2
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            view.getLocationOnScreen(locationOnScreen)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ I am afraid that we are already doing this operation when we want to detect the view position on screen. We should make sure we only do this once and re - use the resolved coordinates as this operation is costly.

Reply:
While it is expensive, in this case it happens only when we encounter the touch override flag on a view so it should be extremely infrequent. We are calling `getLocationOnScreen` in multiple places in Session Replay so refactoring to consolidate these calls for every traversal would be beneficial but I think out of scope for this pr. I can open a ticket to look into this.